config/ConfigParse.cpp

```diff
@@ -0,0 +1,229 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ConfigParse.cpp                                    :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: siun <siun@student.42.fr>                  +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/05/17 22:37:44 by siun              #+#    #+#             */
+/*   Updated: 2024/05/23 18:58:56 by siun             ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "ConfigParse.hpp"
+
+std::string openFile(std::string path)
+{
+	std::ifstream	file(path.c_str());
+	if (!file.is_open())
+	{
+		throw std::runtime_error("Could not open file " + path);
+	}
+	std::string	config((std::istreambuf_iterator<char>(file))
+		, std::istreambuf_iterator<char>());
+	file.close();
+	return config;
+}
+
+std::vector<std::pair<std::string, int> >	findIndent(std::string str)
+{
+	std::vector<std::pair<std::string, int> > configs;
+	for (int i = 0; i < str.size(); i ++)
+	{
+		int start = i;
+		for (start; str[start] == '\t'; start ++)
+			;
+		int	indent = start - i;
+		i = start;
+		int	end = i;
+		for (end; str[end] != '\n'; end ++)
+			;
+		i = end;
+		if (start != end)
+			configs.push_back(std::make_pair(str.substr(start, end - start), indent));
+	}
+	return configs;
+}
+
+std::vector<std::vector<std::pair<std::string, int> > > findChunck(std::vector<std::pair<std::string, int> > indents)
+{
+	std::vector<std::vector<std::pair<std::string, int> > > multiChunck;
+	int start = 0;
+	int	end = 0;
+	
+	for (int i = 0; i < indents.size(); i ++)
+	{
+		start = i;
+		for (i; i < indents.size() && indents[i].first != "server"; i ++)
+			;
+		end = i;
+		if (start != end)
+		{
+			std::vector<std::pair<std::string, int> > chunck(indents.begin() + start, indents.begin() + end);
+			multiChunck.push_back(chunck);
+		}
+	}
+	return multiChunck;
+}
+
+std::string	parseString(const std::vector<std::pair<std::string, int> > chunck, std::string keyword)
+{
+	for (int i = 0; i < chunck.size(); i ++)
+	{
+		if (chunck[i].first.find(keyword) != std::string::npos)
+		{
+			std::string str = chunck[i].first;
```

Comment:
is the variable declaration inside this loop (limiting its scope to the loop) intentional?
also assigning it on declaration recreates them on every loop iteration.

Reply:
yeah I did it on that way, idk why but when I did it other way using less init--declare it on outside, it gave me grammar error.